node/nogossip.go

```diff
@@ -66,11 +66,17 @@ func (n *Node) txAnnStreamHandler(s network.Stream) {
 	// while we were fetching it
 
 	// store in mempool since it was not in tx index and thus not confirmed
-	n.mp.Store(txHash, rawTx)
-	fetched = true
+	ctx := context.Background()
+	if err := n.ce.CheckTx(ctx, rawTx); err != nil {
+		n.log.Warnf("tx %v failed check: %v", txHash, err)
+	} else {
+		n.mp.Store(txHash, rawTx)
+
+		// re-announce
+		go n.announceTx(context.Background(), txHash, rawTx, s.Conn().RemotePeer())
+	}
 
-	// re-announce
-	go n.announceTx(context.Background(), txHash, rawTx, s.Conn().RemotePeer())
+	fetched = true
```

Comment:
Interesting problem here.  If checkTx fails and we don't store it, then it remains in the prefetch status forever.  We need this `fetched = true` only in the case where we store it.  If it reannounces we'll just have to refetch and recheck, but we don't want the prefetch map to fill or reject txns that might be valid later.

Reply:
umm, I was thinking, if we should store rejected txs (basically all the seen transactions) in a cache (FIFO)

Reply:
It might not be rejected forever though.  If a node somehow gets a tx with a none that is too high, it might be valid again later.  For example, a node starts at an arbitrary time that causes it to miss a predecesor txn, or if user broadcasts man quickly and one tx takes a longer gossip route than another.

Reply:
```
	ctx := context.Background()
	if err := n.ce.CheckTx(ctx, rawTx); err != nil {
		n.log.Warnf("tx %v failed check: %v", txHash, err)
	} else {
		n.mp.Store(txHash, rawTx)
		fetched = true

		// re-announce
		go n.announceTx(context.Background(), txHash, rawTx, s.Conn().RemotePeer())
	}
``` 
How about this?

Reply:
umm, but this won't fetch other transactions :/

Reply:
wdym by "other"?